grails-app/views/productAssociation/list.gsp

```diff
@@ -109,7 +109,7 @@
                                         </td>
 
                                         <td>
-                                            <g:link controller="product" action="edit" id="${productAssociationInstance?.product?.id}">
+                                            <g:link controller="product" action="edit" id="${productAssociationInstance?.associatedProduct?.id}">
```

Comment:
shouldn't this have a fallback for `product.id` if `associatedProduct.id` is null?